### PR TITLE
Snapshot compare

### LINKED ISSUE #3977: Development work to plug two selected snapshots into flowviewer for comparison of changes

### PR DESCRIPTION
**Pull request by Steve-Mcl**
closes #3977 

## Description

- updates `frontend/src/components/dialogs/AssetDetailDialog.vue` to support snapshot comparison
- updates flow-renderer dep to 0.4.0
- updates/adds e2e tests

### TODO
- [ ] check and update docs if appropriate

## Related Issue(s)

#3977 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Review by cstns — CHANGES_REQUESTED** [review]
Looks good overall, minor improvements needed

File: package-lock.json [hunk 1]
```diff
@@ -125,6 +125,15 @@
                 "node": ">=18.x"
             }
         },
+        "../../../flow-renderer": {
```
[the remark this answers is not in the document]
Comment (knolleary) [reply package-lock.json:128]: Pretty sure this needs fixing as it won't pass `npm ci`.. but I could be wrong.

**Comment by cstns** [comment]
I can make a quick PR for the changes and implement them myself, it shouldn't take long. If we're in a hurry we can move past them

**Comment by knolleary** [comment]
@cstns the main hurry is its release day tomorrow and this is a headline feature. Please coordinate with @Steve-Mcl on whether we can get this done today.

File: package-lock.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (Steve-Mcl) [reply package-lock.json:128]: that is odd. I did (for dev purposes) install to local package on filesystem but I did `npm uninstall ` then `npm install flow-@flowfuse/renderer@latest` before publishing.

Will fix this

[the remark this answers is not in the document]
Comment (Steve-Mcl) [reply package-lock.json]: resolved in [update package lock to get rid of artifact](https://github.com/FlowFuse/flowfuse/pull/3978/commits/c0a8786f3f1f96e1ae3af27fd8b48865daec4911)
[c0a8786](https://github.com/FlowFuse/flowfuse/pull/3978/commits/c0a8786f3f1f96e1ae3af27fd8b48865daec4911)

**Comment by Steve-Mcl** [comment]
@cstns from our conversation along the lines of a better DX and more "vue way" of doing these things I am happy to see a PR to address those (and i promise to take notes for future similar tasks). However, in the mean time (assuming functionality is as expected) can we push this through?

**Review by cstns — APPROVED** [review]

**Pull request merged**